src/emulate.c

```diff
@@ -379,8 +380,12 @@ static uint32_t peripheral_update_ctr = 64;
     {                                                                 \
         IIF(RV32_HAS(SYSTEM))(ctr++;, ) cycle++;                      \
         code;                                                         \
-    nextop:                                                           \
-        PC += __rv_insn_##inst##_len;                                 \
+        IIF(RV32_HAS(SYSTEM))                                         \
+        (                                                             \
+            if (need_handle_signal) {                                 \
+                need_handle_signal = false;                           \
+                return true;                                          \
```

Comment:
Add more comments, such as "clear pending signals."

Reply:
> Add more comments, such as "clear pending signals."

Introduce a macro, `CLEAR_PENDING_SIGNAL()` which is self-explanatory.